--- a/libtommath/bn_mp_div_2d.c	2019-06-11 16:42:53.774589344 +0200
+++ b/libtommath/bn_mp_div_2d.c	2019-06-11 16:44:03.851160860 +0200
@@ -16,6 +16,8 @@
  */
 
 /* shift right by a certain bit count (store quotient in c, optional remainder in d) */
+#pragma GCC push_options
+#pragma GCC optimize("O1")
 int mp_div_2d (mp_int * a, int b, mp_int * c, mp_int * d)
 {
   mp_digit D, r, rr;
@@ -79,6 +81,7 @@
   mp_clamp (c);
   return MP_OKAY;
 }
+#pragma GCC pop_options
 #endif
 
 /* ref:         $Format:%D$ */
