diff -ruN stress-ng-0.09.47.orig/stress-nop.c stress-ng-0.09.47/stress-nop.c
--- stress-ng-0.09.47.orig/stress-nop.c	2018-11-21 18:06:08.000000000 +0100
+++ stress-ng-0.09.47/stress-nop.c	2019-10-30 10:16:29.702811684 +0100
@@ -26,7 +26,7 @@
 
 #if defined(HAVE_ASM_NOP)
 
-#define NOP	__asm__ __volatile__("nop;")
+#define NOP	__asm__ __volatile__("nop\n;;")
 #define NOP8	NOP;  NOP;  NOP;  NOP;  NOP;  NOP;  NOP;  NOP;
 #define NOP64	NOP8; NOP8; NOP8; NOP8; NOP8; NOP8; NOP8; NOP8;
 
